curated_transformers/generation/dolly_v2.py

```diff
@@ -44,11 +45,15 @@ def from_hf_hub(
         *,
         name: str,
         revision: str = "main",
-        device: Optional[torch.device] = None
+        device: Optional[torch.device] = None,
+        quantization_config: Optional[BitsAndBytesConfig] = None,
     ) -> Self:
         tokenizer = GPTNeoXTokenizer.from_hf_hub(name=name, revision=revision)
         causal_lm = GPTNeoXCausalLM.from_hf_hub(
-            name=name, revision=revision, device=device
+            name=name,
+            revision=revision,
+            device=device,
+            quantization_config=quantization_config,
```

Comment:
I like naming this `quantization_config`, once we support other quantizers, the type could be a union and we do the right thing based on what kind of config struct type is passed. Nicely avoids having many implementation-specific parameters.